### PR TITLE
Namespaces: GetValuebyNamespace

### PR DESCRIPTION
**Pull request by marcin-krolik**
Added function which recursively looks for value represented in namespace in struct composition

### CODE REVIEW

File: ns/namespace.go [hunk 1]
```diff
@@ -177,4 +178,61 @@ func FromCompositionTags(object interface{}, current string, namespace *[]string
...
+	for _, field := range fields {
+		tag, err := reflections.GetFieldTag(object, field, "json")
+		if err != nil {
+			fmt.Printf("Could not find tag for field{%s}\n", field)
+			return nil
+		}
+		// remove omitempty from tag
+		tag = strings.Replace(tag, ",omitempty", "", -1)
+		if tag == current {
+			val, _ := reflections.GetField(object, field)
```
Comment (andrzej-k) [inline ns/namespace.go]: consider not suppressing but rather handling error that can be returned by GetField

**Comment by andrzej-k** [comment]
Please update date in license header -> 2015-2016

File: ns/namespace.go [hunk 2]
```diff
@@ -177,4 +178,61 @@ func FromCompositionTags(object interface{}, current string, namespace *[]string
...
+				idx, _ := strconv.Atoi(ns[1])
+				val := reflect.ValueOf(val)
+				if val.Index(idx).Kind() == reflect.Struct {
+					return GetValueByNamespace(val.Index(idx).Interface(), ns[2:])
+				} else {
+					return val.Index(idx).Interface()
+				}
+			case reflect.Map:
+				key := ns[1]
+				// try uint64 map (memory_stats case)
```
Comment (andrzej-k) [inline ns/namespace.go]: This comment is not clear to me - what's memory_stats case?

File: ns/namespace.go [hunk 3]
```diff
@@ -177,4 +178,61 @@ func FromCompositionTags(object interface{}, current string, namespace *[]string
...
+				if vi, ok := val.(map[string]uint64); ok {
+					return vi[key]
+				}
+				// try with hugetlb map (hugetlb_stats case)
```
Comment (andrzej-k) [inline ns/namespace.go]: This comment is also not very clear

File: ns/namespace.go (hunk 1 above)
Comment (marcin-krolik) [inline ns/namespace.go]: Sure

File: ns/namespace.go (hunk 2 above)
Comment (marcin-krolik) [inline ns/namespace.go]: This should be removed. Leftover from some old implementation for docker collector

File: ns/namespace.go (hunk 3 above)
Comment (marcin-krolik) [inline ns/namespace.go]: This should be removed. Leftover from some old implementation for docker collector

**Comment by marcin-krolik** [comment]
Comments applied

**Comment by andrzej-k** [comment]
LGTM

**Pull request merged**